designs/disruption-controls-by-action.md

```diff
@@ -0,0 +1,218 @@
+# Disruption Controls By Action 
+
+## Motivation
+Users want the ability to have different disruption controls for the budgets. The AKS Provider Drives NodeImageUpgrade, and K8sVersionUpgrade through Drift. Users may want to specify different disruption settings for upgrade, since upgrade is a fundementally different type of disruption in comparison to consolidation for example. Since upgrade changes the behavior of the cluster, its disruption schedule may be different from consolidation.
```

Comment:
Suggestion: It might be worth framing this motivation in terms of user stories to better crystalize how users might actually utilize this new functionally. For example:
1. I want to ensure that Kubernetes auto-upgrades do not occur on block days or outside of working hours, but I'd like to constantly ensure cost savings (e.g. consolidation)
...

Reply:
Sounsd good was thinking similar things.